javascript-source/core/permissions.js

```diff
@@ -208,61 +220,89 @@
      * @returns {boolean}
      */
     function isAdmin(username) {
-        return getUserGroupId(username.toLowerCase()) <= 1 || isOwner(username);
+        return queryDBPermission(username.toLowerCase()) <= PERMISSION.Admin || isOwner(username);
+    }
+
+    /**
+     * @function isModNoTags
+     * @export $
+     * @param {string} username
+     * @returns {boolean}
+     */
+    function isModNoTags(username) {
+        return isModeratorCache(username.toLowerCase()) || queryDBPermission(username.toLowerCase()) <= PERMISSION.Mod || isOwner(username);
+    }
+
+    function checkTags(tags) {
+        return tags !== null && tags != '{}' && tags != '-1' && tags !== undefined;
     }
 
     /**
      * @function isMod
      * @export $
      * @param {string} username
+     * @param {Object} tags
      * @returns {boolean}
      */
-    function isMod(username) {
-        return isModeratorCache(username.toLowerCase()) || getUserGroupId(username.toLowerCase()) <= 2 || isOwner(username);
+    function isMod(username, tags) {
+        if (checkTags(tags) && tags.get('user-type').length() > 0) { // Broadcaster should be included here.
+            return true;
+        }
+
+        $.consoleDebug('Used isMod without tags::' + tags);
+        return isModNoTags(username);
     }
 
     /**
-     * @function isModv3
+     * @function isMod
+     * @deprecated
+     *
      * @export $
      * @param {string} username
      * @param {Object} tags
      * @returns {boolean}
      */
     function isModv3(username, tags) {
```

Comment:
Function is a pass-through to `isMod` with same signature, can be removed and re-mapped